api/refresh_token_logic.py

```diff
@@ -0,0 +1,69 @@
+from flask import request
+from user_agents import parse
+import requests
+import hashlib
+import time
+import os
+from typing import Optional
+from database.postgres.dto.device_info import DeviceInfoDTO
+
+class UserInfoService:
+    def __init__(self, access_tokens_dal, refresh_token_dal):
+        self._access_token_dal = access_tokens_dal
+        self._refresh_token_dal = refresh_token_dal
+        
+    def get_user_device(self) -> str:
+        user_agent = request.headers.get("User-Agent", "")
+        parsed_agent = parse(user_agent)
+
+        device_info = DeviceInfoDTO(
+            browser=f"{parsed_agent.browser.family} {parsed_agent.browser.version_string}",
+            os=f"{parsed_agent.os.family} {parsed_agent.os.version_string}",
+            device=parsed_agent.device.family
+        )
+        return f"{device_info.device} | {device_info.os} | {device_info.browser}"
```

Comment:
Why don't you override __str__ method of DeviceInfoDTO to return `f"{device_info.device} | {device_info.os} | {device_info.browser}"`? 
That way you can just do `return str(device_info)`